### PR TITLE
Feature/forget pw

### PR DESCRIPTION
**Pull request by stanleylhs**

### CODE REVIEW

**Comment by tkbeili** [comment]
Please rebase before I can code review.

**Comment by stanleylhs** [comment]
@tkbeili I didn't know #4 and #6 was in. So they have exactly the same two conflicts. Now solved.

**Comment by tkbeili** [comment]
@stanleylhs  Please add specs for the reset password controller

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  def new
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+    if user
+      user.send_password_reset 
+      @reset_url = edit_password_reset_url(user.password_reset_token)
+      # TODO: send email
+      redirect_to root_url, :notice => "Email sent with password reset instructions."
```
Comment (tkbeili) [inline app/controllers/password_resets_controller.rb:12]: no need to use a hash rocket in here. Also, maybe use `_path` as you're still inside your application

File: app/controllers/password_resets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+class PasswordResetsController < ApplicationController
+  before_action :find_by_reset_token, only: [:edit, :update]
+  def new
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+    if user
+      user.send_password_reset 
+      @reset_url = edit_password_reset_url(user.password_reset_token)
```
Comment (tkbeili) [inline app/controllers/password_resets_controller.rb:10]: Where is this instance variable used?

File: app/controllers/password_resets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  private
+  def find_by_reset_token
+    @user = User.find_by_password_reset_token!(params[:id])    
+  end
+
+  def user_params
+      params.require(:user).permit(:password, :password_confirmation)
+    end
```
Comment (tkbeili) [inline app/controllers/password_resets_controller.rb:43]: Please fix indentations

File: app/controllers/password_resets_controller.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  before_action :find_by_reset_token, only: [:edit, :update]
+  def new
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+    if user
+      user.send_password_reset 
+      @reset_url = edit_password_reset_url(user.password_reset_token)
+      # TODO: send email
```
Comment (tkbeili) [inline app/controllers/password_resets_controller.rb:11]: Seems like you have the mailer set up so please use it :)

File: app/views/password_resets/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<%= form_for @user, :url => password_reset_path(params[:id]) do |f| %>
```
Comment (tkbeili) [inline app/views/password_resets/edit.html.erb:1]: Please use simple form

File: config/routes.rb [hunk 6]
```diff
@@ -1,8 +1,13 @@
 Rails.application.routes.draw do
+  get 'password_resets/new'
```
Comment (tkbeili) [inline config/routes.rb:2]: Do you need this? you have `resources :password_resets` below.

File: config/routes.rb [hunk 7]
```diff
@@ -1,8 +1,13 @@
...
+
   resources :users
   resources :sessions, only: [:new, :create] do
     delete :destroy, on: :collection
   end
+
+  resources :password_resets
```
Comment (tkbeili) [inline config/routes.rb:9]: Please restrict this to only the actions you need

File: app/controllers/password_resets_controller.rb (hunk 1 above)
Comment (stanleylhs) [inline app/controllers/password_resets_controller.rb:12]: Sorry that was just straight copy and paste from rails cast, and the url is my own debugger.

File: app/controllers/password_resets_controller.rb (hunk 2 above)
Comment (stanleylhs) [inline app/controllers/password_resets_controller.rb:10]: Sorry, the instance is just my own debugger, I forget to del it before push

**Pull request closed without merging**